app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -16,16 +16,156 @@
 
 package com.google.droidjump;
 
+import static androidx.navigation.Navigation.findNavController;
+import android.annotation.SuppressLint;
 import android.content.Context;
+import android.graphics.BitmapFactory;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.view.MotionEvent;
+import android.view.SurfaceHolder;
 import android.view.SurfaceView;
 
 public class GameView extends SurfaceView implements Runnable {
 
+    private boolean isPlaying;
+    private int screenX;
+    private int screenY;
+    private int screenMargin;
+    private SurfaceHolder surfaceHolder;
+    private MainActivity activity;
+    private Droid droid;
+    private int timePoint;
+    Thread thread;
+    private int levelTimePoints;
+    private int levelSpeed;
+
     public GameView(Context context) {
         super(context);
+        isPlaying = false;
+    }
+
+    public GameView(Context context, int screenX, int screenY, boolean isPlaying) {
+        super(context);
+        timePoint = 0;
+        surfaceHolder = getHolder();
+        activity = (MainActivity) context;
+        this.screenX = screenX;
+        this.screenY = screenY;
+        this.isPlaying = isPlaying;
+
+        // Margin in px
+        screenMargin = (int) getResources().getDimension(R.dimen.fab_margin);
+        // Measuring droid jump height
+        DrawableElement palm = new DrawableElement(/* x= */0, /* y= */0,
+                /* bitmap= */BitmapFactory.decodeResource(getResources(), R.mipmap.palm));
+        int jumpHeight = palm.getHeight() + 100;
+        // Create droid
+        droid = new Droid(screenMargin, screenY - screenMargin, jumpHeight, getResources());
     }
 
     @Override
     public void run() {
+        receiveLevelDetails();
+        draw();
+        while (isPlaying) {
+            update();
+            draw();
+            sleep();
+            timePoint++;
+        }
+    }
+
+    private void receiveLevelDetails() {
+        // TODO: Serialize current level data and put it in some container
+        levelTimePoints = 200;
+        levelSpeed = 3;
+    }
+
+    public void update() {
+        // TODO: Check if time point is in level data and add data to some container, than move
+        //  it to left
+        if (droid.isJumping()) {
+            if (droid.getY() < droid.getInitialY() - droid.getJumpHeight()) {
+                droid.setJumping(false);
+            } else {
+                // Setting 4th droid character(droid in jump)
+                droid.setBitmap(droid.getDroidTypes()[GameConstants.DROID_JUMPING_CHARACTER_INDEX]);
+                droid.setY(droid.getY() - droid.getJumpHeight() / 3);
+            }
+        } else if (droid.getY() == droid.getInitialY()) {
+            // Droid Animating
+            if (timePoint % 5 < 2) {
```

Comment:
This means that the first step is on 2 out of 5 possible steps, while the second step is on 3 steps. Is it intentional that one step is longer?

Reply:
Can replace "5" with "4" so there will be only 4 possible ones: 0,1,2,3.